### PR TITLE
fix: Introduce case insensitivity in the generator

### PR DESCRIPTION
**Pull request by marikaner**
## Context

If there are entities that have the same name but different casing, we treat them as different names. We did not consider, that the generated names are also the names of the files and most file systems are case insensitive. This PR introduces case insensitive checks in the unique name finder, that are applied for entities and complex types.

Solves https://github.com/SAP/cloud-sdk/issues/560.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [x] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

**Review by jjtang1985 — APPROVED** [review]
LGTM. only some questions regarding the scope of the PR.

File: packages/generator/src/service-name-formatter.ts [hunk 1]
```diff
@@ -120,9 +123,12 @@ export class ServiceNameFormatter {
       ''
     );
 
-    const newName = this.finderServiceWide.findUniqueName(transformedName);
+    const newName = this.finderServiceWide.findUniqueName(
```
[the remark this answers is not in the document]
Comment (marikaner) [reply packages/generator/src/service-name-formatter.ts:126]: What do you mean? I only added a check here that looks at the given names independently of the case.

[the remark this answers is not in the document]
Comment (jjtang1985) [reply packages/generator/src/service-name-formatter.ts:126]: I mean the second parameter used here is `false` (caseSensitive), so we treat `ABCService` and `abcservice` the same.

[the remark this answers is not in the document]
Comment (jjtang1985) [reply packages/generator/src/service-name-formatter.ts:126]: discussed.

**Pull request merged**